content/guides/migrating-to-cypress/protractor.md

```diff
@@ -0,0 +1,872 @@
+
+<Alert type="info">
+
+For more information, check out our [official documentation on debugging](/guides/guides/debugging#Using-debugger)!
+
+</Alert>
+
+## Continuous Integration
+
+Cypress makes it easy to [run your tests in your Continuous Integration environment](/guides/continuous-integration/introduction).
+
+Check out our guides to run your Cypress tests in a [GitHub Action](/guides/continuous-integration/github-actions), [CircleCI](/guides/continuous-integration/introduction#CircleCI), [GitLab CI](/guides/continuous-integration/gitlab-ci), [Bitbucket Pipeline](/guides/continuous-integration/bitbucket-pipelines), or [AWS CodeBuild](/guides/continuous-integration/aws-codebuild).
+
+### Gain Visibility in Headless Mode with Screenshots and Videos
+
+Running browser tests in headless mode (locally or in continuous integration pipeline) can be a bit of a black-box without much visibility. When tests fail, error messages by themselves can often fall short in painting the picture of **why** something failed, especially if assertions were not explicit enough or too indirect. To understand the reason behind test failures it also helps to see the state of the app UI at the point of failure or see the events that led up to the failure.
+
+Cypress assists with debugging in headless mode, by automatically taking a screenshot of the app UI and command log at the exact point of test failure. To help see everything that happened prior to test failure, Cypress provides a video recording (as an MP4 file) of a full test spec run by default.
+
+## Parallelization
+
+One of the worst things that can happen to a developer is to be forced to wait for a 2 hour end-to-end test suite to before verifying something works or not. Instead, the [Cypress Dashboard Service](/guides/dashboard/introduction) allows your tests to run in parallel. If your longest test only test a minute to run, this means that you've just cut down your testing by over 12,000%!
+
+This feature is available in Protractor and requires you to configure your application with multiple options (i.e., `sharedTestFiles`, `maxInstances`, etc.). With Protractor your tests can be parallelized on a per-test basis. With Cypress, your tests can be [parallelized on a per spec file basis](https://docs.cypress.io/guides/guides/parallelization). This is an important distinction between Protractor and Cypress parallelization. One of the reasons why Cypress parallelizes tests per file because some users may write tests that build up state that subsequent tests, although we [do not recommend relying on the state of previous tests](/guides/references/best-practices#Having-tests-rely-on-the-state-of-previous-tests).
```

Comment:
### ### 
```suggestion
This feature is available in Protractor and requires you to configure your application with multiple options (i.e., `sharedTestFiles`, `maxInstances`, etc.). With Protractor your tests can be parallelized on a per-test basis. With Cypress, your tests can be [parallelized on a per spec file basis](https://docs.cypress.io/guides/guides/parallelization). This is an important distinction between Protractor and Cypress parallelization. One of the reasons why Cypress parallelizes tests per file is because some users may write tests that build up state that subsequent tests, although we [do not recommend relying on the state of previous tests](/guides/references/best-practices#Having-tests-rely-on-the-state-of-previous-tests).
```